src/engines/html5/html5.js

```diff
@@ -317,7 +317,12 @@ export default class Html5 extends FakeEventTarget implements IEngine {
       this._eventManager.listen(mediaSourceAdapter, CustomEventType.ABR_MODE_CHANGED, (event: FakeEvent) => this.dispatchEvent(event));
       this._eventManager.listen(mediaSourceAdapter, CustomEventType.TEXT_CUE_CHANGED, (event: FakeEvent) => this.dispatchEvent(event));
       this._eventManager.listen(mediaSourceAdapter, CustomEventType.TRACKS_CHANGED, (event: FakeEvent) => this.dispatchEvent(event));
-      this._eventManager.listen(mediaSourceAdapter, CustomEventType.FRAG_LOADED, (event: FakeEvent) => this.dispatchEvent(event));
+      this._eventManager.listen(mediaSourceAdapter, CustomEventType.FRAG_LOADED, (event: FakeEvent) => {
+        this.dispatchEvent(event);
+        if (this.isLive()) {
+          this.dispatchEvent(new FakeEvent(Html5EventType.DURATION_CHANGE));
```

Comment:
already fired natively by the engine (in hlsjs at least )

Reply:
In shaka this is not fired.

Reply:
so move it to the adapter where is needed